### DISCUSSION TITLE
Integration: DeepSource

### DISCUSSION THREAD

**Discussion by withshubh** [saltstack/salt#67449] [Feature Requests]
I ran DeepSource analysis on my fork of this repository and found some code quality issues. Have a look at the issues caught in this repository by DeepSource [here](https://deepsource.io/gh/withshubh/salt/issues/).

[DeepSource](https://deepsource.io/) is a code review automation tool that detects code quality issues and helps you to automatically fix some of them. You can use DeepSource to track test coverage, Detect problems in Dockerfiles, etc. in addition to detecting issues in code. 

The PR #59385 fixed some of the issues caught by DeepSource.

All the features of the DeepSource are mentioned [here](https://deepsource.io/features/).
I'd suggest you integrate DeepSource since it is free for Open Source projects forever.

Integrating DeepSource to continuously analyze your repository:
- Install DeepSource on your repository [here](https://deepsource.io/signup).
- Create `.deepsource.toml` configuration specific to this repo or use the configuration mentioned below which I used to run the analysis on the fork of this repo.
- Activate analysis [here](https://deepsource.io/gh/saltstack/salt/).

```
version = 1

test_patterns = ["tests/**/test_*.py"]

exclude_patterns = [
    "/templates/**",
    "/changelog/**",
    "/cicd/**",
    "/conf/**",
    "/rfcs/**"
]

[[analyzers]]
name = "python"
enabled = true

  [analyzers.meta]
  runtime_version = "3.x.x"
```

**Comment by welcome** [comment]
Hi there! Welcome to the Salt Community! Thank you for making your first contribution. We have a lengthy process for issues and PRs. Someone from the Core Team will follow up as soon as possible. In the meantime, here’s some information that may help as you continue your Salt journey.
Please be sure to review our [Code of Conduct](https://github.com/saltstack/salt/blob/master/CODE_OF_CONDUCT.md). Also, check out some of our community resources including:

  - [Community Wiki](https://github.com/saltstack/community/wiki)
  - [Salt’s Contributor Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html)
  - [Join our Community Slack](https://saltstackcommunity.herokuapp.com/)
  - [IRC on Freenode](https://webchat.freenode.net/#salt)
  - [SaltStack YouTube channel](https://www.youtube.com/user/SaltStack)
  - [SaltStackInc Twitch channel](https://www.twitch.tv/saltstackinc)

There are lots of ways to get involved in our community. Every month, there are around a dozen opportunities to meet with other contributors and the Salt Core team and collaborate in real time. The best way to keep track is by subscribing to the Salt Community Events Calendar.
If you have additional questions, email us at core@saltstack.com. We’re glad you’ve joined our community and look forward to doing awesome things with you!

**Comment by withshubh** [comment]
Hey @s0undt3ch 👋 

Any update on this?

**Comment by s0undt3ch** [comment]
At the moment we're not considering adding yet another tool to our toolset, even though it did provide valuable information.

We might revisit this if we ever decide to revisit the available options.

Thank You.
